fritz_interview.py

```diff
@@ -0,0 +1,70 @@
+# Lucky Dollar Store Coding Challenge
+import random
+names = ["Eric", "Chris", "Dan", "Jameson"]
+		
+
+def record_customer_purchase(customer):
+     # Fill this in
+     print("- Added new raffle ticket for \"%s\"" % customer)
+     return customer
+
+
+def can_win(customer, pool):
+	if not pool:  
+		return True
+	elif customer in pool.keys():
+		if pool[customer] == 0:
+			return True
+	elif customer not in pool.keys():
+		return True
+	return False
+
+
+def run_weekly_raffle(num_winners, num_purchases):
+	# pick winner(s)
+	pool = []
+	winning_customer = []
+	for n in range(0, num_purchases):
+		customer_index = random.randrange(0, len(names))
+		pool.append(record_customer_purchase(names[customer_index]))
+	
+	for i in range(0, num_winners):
+		winner_index = random.randrange(0, len(pool))
+		winner = pool[winner_index]
+		winning_customer.append(winner)
+	return winning_customer
+
+def main():
+	pool = []
+	num_drawings = 6
+	cant_win_again = {}
+	for i in range(1, num_drawings + 1):	
+		print("Raffle Drawing %d" % i)
+		num_purchases = random.randrange(1,5)
+		winners = list(set(run_weekly_raffle(2, num_purchases))) # same winner can't win in same drawing
+		print("- Picking the weekly winner….", end = " ")
+		# how many winners to print
+		if len(winners) == 1 and can_win(winners[0], cant_win_again):
+			print(winners[0])
+			cant_win_again.update({winners[0]:3})
+		elif len(winners) > 1:
+			# for j in range(0, len(winners)-1):
+			# 	if can_win(winners[j]):
+			# 		pass
+			# 	print(winners[j], end = " & ")
+			# 	#winners[j].win()
+			# print(winners[len(winners) - 1])
+			for winner in winners:
+				if can_win(winner, cant_win_again):
+					print(winner)
+					cant_win_again.update({winner:3})
+		else:
```

Comment:
If there are no eligible winners in the pool, it would still be nice to show this message that no one won.

Here's an example where in raffle 5, Dan, Chris, and Jameson are all ineligible to win.
```
Raffle Drawing 5
- Added new raffle ticket for "Dan"
- Added new raffle ticket for "Chris"
- Added new raffle ticket for "Jameson"
- Added new raffle ticket for "Dan"
- Picking the weekly winner…. Raffle Drawing 6
- Added new raffle ticket for "Eric"
- Picking the weekly winner…. No one :(
```

Reply:
New output:
`- Added new raffle ticket for "Hally"
- Added new raffle ticket for "Jerry"
- Added new raffle ticket for "Bill"
- Added new raffle ticket for "Roland"
Raffle Drawing 1
- Picking the weekly winner….
Hally
Bill
Raffle Drawing 2
- Picking the weekly winner….
Roland
Raffle Drawing 3
- Picking the weekly winner….
No one :(
Raffle Drawing 4
- Picking the weekly winner….
No one :(
Raffle Drawing 5
- Picking the weekly winner….
Jerry
Raffle Drawing 6
- Picking the weekly winner….
Roland
Raffle Drawing 7
- Picking the weekly winner….
No one :(`